src/storage/jni.rs

```diff
@@ -0,0 +1,320 @@
+use jni::JNIEnv;
+use jni::objects::{JClass, JValue, JString, JObject};
+use rocksdb::ColumnFamily;
+use jni::sys::{jobject, jlong, jboolean, jbyteArray, jobjectArray, JNI_TRUE, JNI_FALSE};
+use std::ptr::null_mut;
+use crate::common::jni::{unwrap_ptr, unwrap_mut_ptr, map_to_java_map, create_java_object, java_array_to_vec_byte, java_map_to_vec_byte};
+use crate::common::storage::ColumnFamiliesManager;
+use crate::common::Reader;
+use crate::common::transaction::TransactionBasic;
+use crate::storage::Storage;
+use crate::storage::transaction::Transaction;
+
+// ------------------------------------- Storage JNI wrappers -------------------------------------
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeOpen(
+    _env: JNIEnv,
+    _class: JClass,
+    _storage_path: JString,
+    _create_if_missing: jboolean
+) -> jobject
+{
+    let storage_path = _env.get_string(_storage_path)
+        .expect("Should be able to read jstring as Rust String");
+
+    if let Ok(storage) = Storage::open(
+        storage_path.to_str().unwrap(),
+        _create_if_missing != 0
+    ){
+        let storage_class = _env.find_class("com/horizen/storage/Storage")
+            .expect("Should be able to find class Storage");
+        create_java_object(&_env, &storage_class, storage)
+    } else {
+        null_mut()
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeClose(
+    _env: JNIEnv,
+    _class: JClass,
+    _storage: *mut Storage,
+)
+{
+    if !_storage.is_null(){
+        drop(unsafe { Box::from_raw(_storage) })
+    }
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_horizen_storage_Storage_nativeGet(
+    _env: JNIEnv,
+    _storage: JObject,
+    _cf: JObject,
+    _key: jbyteArray
+) -> jbyteArray
+{
+    let storage = unwrap_ptr::<Storage>(&_env, _storage);
+    let cf = unwrap_ptr::<ColumnFamily>(&_env, _cf);
+
+    let key = _env.convert_byte_array(_key)
+        .expect("Should be able to convert to Rust byte array");
+
+    if let Some(value) = storage.get_cf(cf, key.as_slice()){
+        _env.byte_array_from_slice(value.as_slice())
+            .expect("Should be able to convert Rust slice into jbytearray")
+    } else {
+        null_mut()
```

Comment:
is `null_mut()` fine from java perspective?
What is the difference between it and `JObject::null()` from jni lib?

Reply:
jobject is actually an internal Rust-pointer, this is its definition:
pub type jobject = *mut _jobject;

so null_mut() works as well (actually the JObject::null() is implemented in the same way: 
/// Creates a new null object
    pub fn null() -> JObject<'a> {
        (::std::ptr::null_mut() as jobject).into()
    }
)
That's why it works. But anyway the JObject::null() method should be used, thanks for pointing it!